static/js/src/advantage/react/hooks/useUsingTestBackend.ts

```diff
@@ -0,0 +1,9 @@
+import type { UsingTestBackend } from "advantage/api/types";
+import { useQuery } from "react-query";
+
+export const useUsingTestBackend = () => {
+  const { data: usingTestBackend } = useQuery<UsingTestBackend>(
+    "usingTestBackend"
+  );
+  return { usingTestBackend };
```

Comment:
How are you accessing the window object here?

Reply:
I'm only accessing the window object in [`useLoadWindowData`](https://github.com/canonical-web-and-design/ubuntu.com/pull/10135/files#diff-934076135df0b5c96a13328ac7f3355537dcfd4dc21aad9d7bede7b8e6377d21R17) so that at every other point in the app (like here and in tests) I can get/add data to/from the store and not worry about setting variables on the window.

Once the API is available I'll be able to remove the window hook and add API calls to these hooks.

Do you think that make sense?

Reply:
Ah ok I see! It might be a bit overkill to have an ad-hoc hook just to access one query?
What I would probably do is just use this in the consuming components
```jsx
  const { data: usingTestBackend } = useQuery<UsingTestBackend>(
    "usingTestBackend"
  );
```

Is there some advantage I'm missing to wrap this in a custom hook? Or maybe it'll make more sense when the data actually comes from an API call

Reply:
Yeah with this really simple flag there's not a huge amount of benefit at the moment except:
a) the type is set in one place and,
b) the data is consumed consistently (window -> window fetch hook -> react query -> hook to get data from react query) so that it's obvious to always use the hook to get the data vs a mix of custom hooks and `useQuery`,
c) the hook is set up ready to add the API calls when they're ready.

Does that sound reasonable?